src/Pos/Wallet/Web/ClientTypes/Types.hs

```diff
@@ -0,0 +1,572 @@
+-- safe to introduce `class PSSimplified` that would have the same
+-- implementation has it is with Buildable Address but then person
+-- will know it will probably change something for purescript.
+-- | Transform Address into CId
+addressToCId :: Address -> CId w
+addressToCId = CId . CHash . sformat F.build
+
+cIdToAddress :: CId w -> Either Text Address
+cIdToAddress (CId (CHash h)) = decodeTextAddress h
+
+encToCId :: EncryptedSecretKey -> CId w
+encToCId = addressToCId . makePubKeyAddress . encToPublic
+
+-- | Client transaction id
+newtype CTxId = CTxId CHash
+    deriving (Show, Eq, Generic, Hashable)
+
+mkCTxId :: Text -> CTxId
+mkCTxId = CTxId . CHash
+
+-- | transform TxId into CTxId
+txIdToCTxId :: TxId -> CTxId
+txIdToCTxId = mkCTxId . sformat hashHexF
+
+convertTxOutputs :: [TxOut] -> [(CId w, CCoin)]
+convertTxOutputs = map (addressToCId . txOutAddress &&& mkCCoin . txOutValue)
+
+-- [CSM-309] This may work until transaction have multiple source accounts
+-- | Get all addresses of source account of given transaction.
+getTxSourceAccountAddresses
+    :: [CWAddressMeta]      -- ^ All addresses in wallet
+    -> NonEmpty (CId Addr)  -- ^ Input addresses of transaction
+    -> Maybe [CId Addr]     -- ^ `Just` addrs if the wallet is source of
+                            --   transaction, `Nothing` otherwise
+getTxSourceAccountAddresses walAddrMetas (someInputAddr :| _) = do
+    someSrcAddrMeta <- find ((== someInputAddr) . cwamId) walAddrMetas
+    let srcAccount = addrMetaToAccount someSrcAddrMeta
+    return $
+        map cwamId $
+        filter ((srcAccount ==) . addrMetaToAccount) walAddrMetas
+
+-- | Produces a function which for a given address says whether this address
+-- belongs to the same account as the addresses which are the sources
+-- of a given transaction. This assumes that the source addresses belong
+-- to the same account.
+-- Note that if you apply this function to the outputs of a transaction,
+-- you will effectively get /change/ addresses.
+isTxLocalAddress
+    :: [CWAddressMeta]      -- ^ All addresses in wallet
+    -> NonEmpty (CId Addr)  -- ^ Input addresses of transaction
+    -> (CId Addr -> Bool)
+isTxLocalAddress wAddrMetas inputs = do
+    let mLocalAddrs = getTxSourceAccountAddresses wAddrMetas inputs
+    case mLocalAddrs of
+       Just changeAddrs -> do
+           -- if given wallet is source of tx, /local addresses/
+           -- can be fetched according to definition
+           let changeAddrsSet = S.fromList changeAddrs
+           flip S.member changeAddrsSet
+       Nothing -> do
+           -- if given wallet is *not* source of tx, then it's incoming
+           -- transaction, and only addresses of given wallet are *not*
+           -- local addresses
+           -- [CSM-309] This may work until transaction have multiple
+           -- destination addresses
+           let nonLocalAddrsSet = S.fromList $ cwamId <$> wAddrMetas
+           not . flip S.member nonLocalAddrsSet
+
+mkCTxs
```

Comment:
This (and other functions in this file) might be moved to a separate module, like `Pos.Wallet.Web.ClientTypes.Functions`